src/System.CommandLine.Hosting/HostingAction.cs

```diff
@@ -13,11 +14,11 @@ internal sealed class HostingAction : BindingHandler
     {
         private readonly Func<string[], IHostBuilder> _hostBuilderFactory;
         private readonly Action<IHostBuilder> _configureHost;
-        private readonly CliAction _actualAction;
+        private readonly AsynchronousCliAction _actualAction;
 
         internal static void SetHandlers(CliCommand command, Func<string[], IHostBuilder> hostBuilderFactory, Action<IHostBuilder> configureHost)
         {
-            command.Action = new HostingAction(hostBuilderFactory, configureHost, command.Action);
+            command.Action = new HostingAction(hostBuilderFactory, configureHost, (AsynchronousCliAction)command.Action);
```

Comment:
This breaks things for me, as the command.Action is either a HelpAction or AnonymousSynchronousCliOption giving me InvalidCastExceptions. Is UseHost() going to be limited to just async actions now?